backend/pkg/api/handlers/auth.go

```diff
@@ -239,9 +292,95 @@ func (h *HandlerService) InternalPostUserConfirm(w http.ResponseWriter, r *http.
 
 	// TODO: purge all user sessions
 
+	returnNoContent(w)
+}
+
+func (h *HandlerService) InternalPutUserPasswordReset(w http.ResponseWriter, r *http.Request) {
+	var v validationError
+	req := struct {
+		Email string `json:"email"`
+	}{}
+	if err := v.checkBody(&req, r); err != nil {
+		handleErr(w, err)
+		return
+	}
+
+	// validate email
+	email := v.checkEmail(req.Email)
+	if v.hasErrors() {
+		handleErr(w, v)
+		return
+	}
+
+	userId, err := h.dai.GetUserByEmail(r.Context(), email)
+	if err != nil {
+		if err == dataaccess.ErrNotFound {
+			returnConflict(w, errors.New("email not registered"))
+		} else {
+			handleErr(w, err)
+		}
+		return
+	}
+
+	// send password reset email
+	err = h.sendResetEmail(r.Context(), userId, email)
+	if err != nil {
+		handleErr(w, err)
+		return
+	}
+
 	returnOk(w, nil)
 }
 
+func (h *HandlerService) InternalPostUserPasswordResetHash(w http.ResponseWriter, r *http.Request) {
+	var v validationError
+	resetToken := v.checkUserEmailToken(mux.Vars(r)["token"])
+	req := struct {
+		Password string `json:"password"`
+	}{}
+	if err := v.checkBody(&req, r); err != nil {
+		handleErr(w, err)
+		return
+	}
+	password := v.checkPassword(req.Password)
+	if v.hasErrors() {
+		handleErr(w, v)
+		return
+	}
+
+	// check token validity
+	userId, err := h.dai.GetUserIdByResetHash(r.Context(), resetToken)
+	if err != nil {
+		handleErr(w, err)
+		return
+	}
+	resetTime, err := h.dai.GetPasswordResetTime(r.Context(), userId)
+	if err != nil {
+		handleErr(w, err)
+		return
+	}
+	if resetTime.Add(authEmailExpireTime).Before(time.Now()) {
+		handleErr(w, errors.New("reset link expired"))
+		return
+	}
+
+	// change password
+	passwordHash, err := bcrypt.GenerateFromPassword([]byte(password), 10)
+	if err != nil {
+		handleErr(w, errors.New("error hashing password"))
+		return
+	}
+	err = h.dai.UpdateUserPassword(r.Context(), userId, string(passwordHash))
+	if err != nil {
+		handleErr(w, err)
+		return
+	}
+
+	// TODO: purge all user sessions
```

Comment:
Pls implement, see https://github.com/alexedwards/scs?tab=readme-ov-file#enumerate-all-sessions

Reply:
Will do in #716 after some testing